### PR TITLE
Add More Extensive ONNX BERT Tests

### PR DESCRIPTION
**Pull request by rayankrish**
Testing Legacy vs Experimental losses and Experimental vs hardcoded losses from Legacy runs.

- LR Scheduler
- Optimizer Parameter groups
- Loss Scaler + Mixed Precision

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 1]
```diff
@@ -17,59 +19,518 @@
...
+        legacy_sample_input = [*sample_input, learning_rate]
+
+        legacy_losses.append(legacy_trainer.train_step(legacy_sample_input).cpu().item())
+
+    _test_helpers.assert_model_outputs(experimental_losses, legacy_losses, True, rtol=1e-4)
+
+@pytest.mark.parametrize("initial_lr, lr_scheduler, legacy_lr_scheduler", [
+    (1.0, optim.lr_scheduler.ConstantWarmupLRScheduler, constantlrscheduler),
+])
+def testToyBERTModelLRSchedulerLegacyExperimental(initial_lr, lr_scheduler, legacy_lr_scheduler):
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:458]:
```suggestion
def testToyBERTModelLegacyExperimentalLRScheduler(initial_lr, lr_scheduler, legacy_lr_scheduler):
```

Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:458]: The prefix must be identical between all tests to allow selective tests like `pytest -sk testToyBERTModelLegacyExperimental*`

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 2]
```diff
@@ -17,59 +19,518 @@
...
+    model_desc = bert_model_description()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:459]:
```suggestion
        # Experimental API
        model_desc = bert_model_description()
```

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 3]
```diff
@@ -17,59 +19,518 @@
...
+
+
+###############################################################################
+# Temporary tests comparing Legacy vs Experimental ORTTrainer APIs ############
+###############################################################################
+
+
+def testToyBERTModelLegacyExperimental():
+    num_batches = 10
+
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:408]:
```suggestion

    # Experimental API
```

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 4]
```diff
@@ -17,59 +19,518 @@
...
+    # Change the hyper parameters for all parameters
+    ([], legacy_optim_params_a),
+    # Change the hyperparameters for a subset of hardcoded parameters
+    ([{'params':["bert.embeddings.LayerNorm.weight", "bert.encoder.layer.0.attention.output.dense.weight"], "alpha": 0.9, "beta": 0.999, "lambda_coef": 0.0, "epsilon": 1e-6}], legacy_optim_params_b),
+    # Change the hyperparameters for a generated set of paramers
+    (optimizer_parameters(load_bert_onnx_model()), legacy_optim_params_c)
+])
+def testToyBERTModelCustomOptimParametersLegacyExperimental(params, legacy_optim_map):
+    total_steps = 10
+
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:515]:
```suggestion
    # Experimental API
```

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 5]
```diff
@@ -17,59 +19,518 @@
...
+    model_desc = bert_model_description()
+    model = load_bert_onnx_model()
+
+    optim_config = optim.LambConfig(params, alpha= 0.9, beta= 0.999, lambda_coef= 0.01, epsilon= 1e-6)
+    optim_config = optim.LambConfig()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:520]:
```suggestion
```
This line was overriding your lamb config

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py [hunk 6]
```diff
@@ -17,59 +19,518 @@
...
+                       loss_scaler = legacy_loss_scaler)
+    legacy_losses = []
+    for i in range(total_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+        legacy_sample_input = [*sample_input, learning_rate]
+
+        legacy_losses.append(legacy_trainer.train_step(legacy_sample_input).cpu().item())
+
+    print(experimental_losses)
+    print(legacy_losses)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py:610]: You should assert results here
